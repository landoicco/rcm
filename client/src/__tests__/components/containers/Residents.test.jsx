import { render, screen } from "@testing-library/react";
import {
  AllResidents,
  ResidentSearchResult,
} from "../../../components/containers/Residents";

describe("AllResidents", () => {
  test("The component AllResidents render correctly", () => {
    const { container } = render(<AllResidents />);
    expect(container).toMatchSnapshot();
  });

  test("Should have only one heading element", async () => {
    render(<AllResidents />);

    // Get all elements with a 'heading' role
    const elements = await screen.findAllByRole("heading");
    expect(elements.length === 1).toBe(true);
  });

  test("The word 'Residents' should be rendered at least once", async () => {
    render(<AllResidents />);

    // Find all elements with the word 'Residents' on it
    const elements = await screen.findAllByText("Residents");
    expect(elements.length > 0).toBe(true);
  });
});

describe("ResidentSearchResult", () => {
  test("The component ResidentSearchResult render correctly", () => {});
});
