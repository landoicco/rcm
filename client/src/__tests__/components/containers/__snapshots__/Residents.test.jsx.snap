// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllResidents The component AllResidents render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h3
      class="center"
    >
      Residents
    </h3>
  </div>
</div>
`;
